### PR TITLE
Add support for normalizing $default:$default

### PR DESCRIPTION
**Pull request by matanlurey**
This is the first part of work needed for https://github.com/dart-lang/build/issues/1075.

After this, `apply_builders.dart` will just need to normalize the key.

(Also ran dartfmt, which looks like it changed again?)

### CODE REVIEW

File: build_config/lib/build_config.dart [hunk 1]
```diff
@@ -13,4 +13,7 @@ export 'src/build_config.dart'
         TargetBuilderConfig,
         TargetBuilderConfigDefaults;
 export 'src/key_normalization.dart'
-    show normalizeBuilderKeyUsage, normalizeTargetKeyUsage;
+    show
+        normalizeBuilderKeyUsage,
+        normalizeTargetKeyUsage,
+        normalizeTargetKeyDefinition;
```
Comment (jakemac53) [inline build_config/lib/build_config.dart]: I think we really just want `normalizeTargetKeyUsage` to understand this as opposed to exposing `normalizeTargetKeyDefinition`? 

The things we actually want to add support for are all usages I believe.

Comment (matanlurey) [reply build_config/lib/build_config.dart]: Ah my bad, let me do that instead :)

Comment (matanlurey) [reply build_config/lib/build_config.dart]: PTAL

File: build_config/lib/src/key_normalization.dart [hunk 2]
```diff
@@ -41,11 +41,17 @@ String normalizeTargetKeyDefinition(String targetKey, String packageName) =>
...
+///   - "$default" => "$packageName:$packageName"
+///   - "$default:$default": => "$packageName:$packageName"
 ///   - "some_package" => "some_package:some_package"
 ///   - ":some_target" => "$packageName:some_target"
 ///   - "some_package:some_target" => "some_package:some_target"
 String normalizeTargetKeyUsage(String targetKey, String packageName) =>
-    _normalizeUsage(targetKey, packageName, ':');
+    targetKey == _defaultTargetNamePlaceholder ||
+            targetKey ==
+                '$_defaultTargetNamePlaceholder:$_defaultTargetNamePlaceholder'
```
Comment (natebosch) [inline build_config/lib/src/key_normalization.dart]: might not be necessary, but do we want to accept `:$default`? We do accept `:my_package` I think

File: build_config/lib/src/key_normalization.dart [hunk 3]
```diff
@@ -41,11 +41,17 @@ String normalizeTargetKeyDefinition(String targetKey, String packageName) =>
 ///
 /// Example normalizations:
 ///
+///   - "$default" => "$packageName:$packageName"
+///   - "$default:$default": => "$packageName:$packageName"
```
Comment (natebosch) [inline build_config/lib/src/key_normalization.dart]: [nit] One extra `:` snuck in before the `=>`

File: build_runner/lib/src/package_graph/apply_builders.dart [hunk 4]
```diff
@@ -157,31 +157,31 @@ Iterable<BuildAction> _createBuildActionsWithinCycle(
...
-              ..addAll(builderConfigOverrides));
-        return new BuildAction(b(options), node.package.name,
-            builderOptions: options,
-            targetSources: node.target.sources,
-            generateFor: generateFor,
-            isOptional: builderApplication.isOptional,
-            hideOutput: builderApplication.hideOutput);
-      }));
+      .where((targetNode) =>
+      _shouldApply(builderApplication, targetNode, applyWith))
```
Comment (natebosch) [inline build_runner/lib/src/package_graph/apply_builders.dart]: did dartfmt go haywire here? Some of this looks off to me.

File: build_config/lib/build_config.dart (hunk 1 above)
Comment (natebosch) [reply build_config/lib/build_config.dart]: I can't find the usage of `normalize*` from apply_builders... did something change?

File: build_config/lib/src/key_normalization.dart (hunk 2 above)
Comment (matanlurey) [reply build_config/lib/src/key_normalization.dart]: PTAL again, thanks @natebosch

File: build_config/lib/src/key_normalization.dart (hunk 3 above)
Comment (jakemac53) [inline build_config/lib/src/key_normalization.dart]: I am not sure if we need to actually support this variant, but I guess I don't see a reason why not either?

**Pull request merged**